mayastor/src/bdev/util/uri.rs

```diff
@@ -54,3 +60,32 @@ pub(crate) fn boolean(
     }
     value.parse::<bool>()
 }
+
+pub(crate) fn uuid(
+    value: Option<String>,
+    url: &Url,
+) -> Result<Option<uuid::Uuid>, NexusBdevError> {
+    Ok(match value {
+        Some(value) => Some(uuid::Uuid::parse_str(&value).context(
+            nexus_uri::UuidParamParseError {
+                uri: url.to_string(),
+                parameter: "uuid".to_string(),
+            },
+        )?),
+        None => None,
+    })
+}
+
+pub(crate) fn dev_set_uuid(name: &str, uuid: Option<uuid::Uuid>) {
+    match Bdev::lookup_by_name(&name) {
+        Some(mut bdev) => {
+            bdev.set_uuid(uuid.map(|u| u.to_string()));
+        }
+        // Should we fail to create a device if we can't set
+        // the UUID? Then we need to return error here
+        None => error!(
```

Comment:
The error here is incorrect, this error would suggest we could not find the bdev.... not that we can not set the uri. The set_uuid() function returns void so it has to be the error from not being able to find it :)

Reply:
yeah, it's more a failed to "find and set". But what happens if we fail to find? Or simply assume that since we've just created it it must have be found?

Reply:
if we fail to find, then we have a bigggg problem ;)